src/pages/index.tsx

```diff
@@ -1,24 +1,48 @@
 import CampaignHero from "../components/hero/campaignHero";
-import {
-  SkipNavigation,
-  TemplateAppContainer,
-} from "@nypl/design-system-react-components";
+import SwimLanes from "../components/swimlanes/swimLanes";
+import { TemplateAppContainer } from "@nypl/design-system-react-components";
+import data from "@/data/lanes";
 
-export default function Home() {
+export default function Home({ lanesWithNumItems }) {
   return (
     <TemplateAppContainer
       aboveHeader={<p> Notification banner </p>}
       header={<p> Header </p>}
       breakout={<CampaignHero />}
-      contentPrimary={
-        <>
-          <p>First swim lane</p>
-          <p>Featured Content</p>
-          <p>Rest of swim lanes</p>
-          <p>Explore further links</p>
-        </>
-      }
+      contentPrimary={<SwimLanes lanesWithNumItems={lanesWithNumItems} />}
       renderSkipNavigation={true}
     />
   );
 }
+
+export async function getServerSideProps() {
+  const lanes = data.lanes;
+  const lanesWithNumItems = await Promise.all(
+    lanes.map(async (lane) => {
+      const updatedCollections = await Promise.all(
+        lane.collections.map(async (collection) => {
+          const apiUrl = `/api/getNumItems/?uuid=${collection.uuid}`;
```

Comment:
This can work if the URL is updated to `http://localhost:3000/api/getNumItems/?uuid=${collection.uuid}`. However, `getServerSideProps` and the internal Next.js API endpoint `/api/getNumItems` are both on the server. Making a server call to a server endpoint is not necessary. When deployed to Vercel or AWS, it won't really know what "http://localhost:3000" is and it's not best practice to include that in API calls. (For a client call to the server endpoint, using `/api/getNumItems?uuid...` is correct).

Recommendation:

Remove the `/api/getNumItems` endpoint and convert the `getNumItems` function so that it returns the item count instead of a `res.status(...).json(..)` response. Then you can call the function here in `getServerSideProps` as `const { numItems } = getNumItems(collection.uuid)`. This function can be moved to a "home page utils" or "collection utils" file in case more related functions are added to the file later on.

If `/api/getNumItems` is needed, you can reuse the logic of the `getNumItems` function in both the API endpoint handler function and in this `getServerSideProps` function.

Reply:
@EdwinGuzman That makes way more sense now thank you